packages/back-end/src/services/datasource.ts

```diff
@@ -139,3 +178,42 @@ export async function testQuery(
     };
   }
 }
+
+export async function createInitialInformationSchema(
+  datasource: DataSourceInterface,
+  organization: string
+) {
+  const { informationSchema } = await generateInformationSchema(datasource);
+
+  // Loop through each database, schema, and table, and create Mongo record for each table's columns.
+  if (informationSchema) {
+    for (const database of informationSchema) {
+      for (const schema of database.schemas) {
+        for (const table of schema.tables) {
+          const result = await createInformationSchemaTable(
+            table.columns,
+            organization,
+            table.table_name
+          );
+          table.id = result.id;
+        }
+      }
```

Comment:
Some performance-related questions/concerns: How many databases is a customer likely to have? How many schemas is a database likely to have? How many tables is a schema likely to have? If a customer has 10 databases, each with 10 schemas, and each with 10 tables, that's 1,000 hits to Mongo. And how many organizations do we have and how often will this run for each organization? Does it make sense to reconsider the data modelling to avoid this kind of load on Mongo? If this will be run infrequently then perhaps it's ok so we can get the functionality out and optimize later if we see any issues.

Reply:
@tinahollygb - Thanks for raising this - I just pushed up a [commit](https://github.com/growthbook/growthbook/pull/940/commits/526d55c77718d3884f207366103fef09d0d25346) to address this.  

With this change, we're now building an array of `tablesToCreate` inside of the loops, and at the end, calling `createInformationSchemaTable` once with that array. Mongo's `create` method can accept either a single record, or an array of records, so this allows us to reduce the number of database calls to a single call for all tables.

In doing so, I also added some properties to each table record for `database`, `schema`, and `table`, so when the front-end is doing the look up for a specific table, we'll use that information, rather than using the `table_id` like we were in the past. This should ensure that we get the correct table data in the event a datasource has multiple databases that have duplicate table names. (We're also passing the organization's ID to further reduce pulling the wrong table, but that's always been the case).

Let me know what you think of this solution - I'll need to do a bit of tidying up, but I just wanted to get this pushed up for feedback.